### PR TITLE
User route

### PR DESCRIPTION
**Pull request by tomerdu**
userRoute - 

usersRouter.get('/', usersController.handleGetUsers);
usersRouter.get('/:email', usersController.handleGetUser);
usersRouter.put('/:email', usersController.handleUpdateUser);
usersRouter.delete('/:email', usersController.handleDeleteUser);

userController is ready !

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module.exports = {
```
Comment (DaveAvigad) [inline .eslintrc.js:1]: 🎉

File: controllers/loginController.js [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+const bcrypt = require('bcrypt');
+const { userExist, statusCheck, validPassword } = require('../services/authService');
+const { generatePassword, sendEmailPassword } = require('../services/authService');
+
+const { User } = require('../services/authService');
+
+const loginControl = async (req, res) => {
+  try {
+    const user = await userExist(req.body.email);
+    await validPassword(req.body.password, user.password);
```
Comment (DaveAvigad) [inline controllers/loginController.js:10]: beautiful code!

File: controllers/usersController.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const { User } = require('../services/authService');
+const { oneTimePass } = require('../services/registerService');
+
+const getUsers = async (req, res) => res.sendStatus(200);
+
+async function handleGetUsers(req, res) {
+  const showAllUser = await User.find({});
```
Comment (DaveAvigad) [inline controllers/usersController.js:7]: `showAllUser` why not just `users`?

File: controllers/usersController.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+async function handleGetUser(req, res) {
+  const user = await User.retrieve({ email: req.params.email });
+  return res.send(JSON.stringify(user));
+}
+
+async function handleUpdateUser(req, res) {
+  try {
+    console.log(req.body);
```
Comment (DaveAvigad) [inline controllers/usersController.js:18]: remove unnecessary logs before production deployment

File: data/MongoStorage.js [hunk 5]
```diff
@@ -2,43 +2,42 @@ const { EventEmitter } = require('events');
...
-    }
-
-    delete (id) {
-        return this.Model.deleteOne({ id });
-    }
-
-    update (id, data) {
-        return this.Model.updateOne({ id }, data);
-    }
+mongoose.set('strictQuery', true);
```
Comment (DaveAvigad) [inline data/MongoStorage.js:5]: hope you also understand this file and not only coPaste it

File: middleware/authenticate.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// const jwt = require('jsonwebtoken');
```
Comment (DaveAvigad) [inline middleware/authenticate.js:1]: do not forget to remove commented-out code before deployment

File: middleware/validator.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    res.set({ authorization: `Bearer ${accessToken}` });
+    req.token = { refreshToken, accessToken: `Bearer ${accessToken}` };
+    next();
+  } catch (e) {
+    return res.sendStatus(401);
+  }
+};
+
+function generateAccessToken(user) {
+  return jwt.sign(user, process.env.ACCESS_TOKEN_SECRET, { expiresIn: '5s' });
```
Comment (DaveAvigad) [inline middleware/validator.js:32]: `expiresIn: '5s'` should be in config file

File: middleware/validator.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+      next();
+    } else {
+      req.token = { refreshToken: req.body.refreshToken, accessToken: authHeader };
+      next();
+    }
+  });
+};
+
+const validation = (req, res, next) => {
+  try {
```
Comment (DaveAvigad) [inline middleware/validator.js:67]: consider using some validator (see https://ajv.js.org/ or https://www.npmjs.com/package/jsonschema)

File: middleware/validatorService.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
+const mailValidator = require('email-validator');
+const passwordValidator = require('password-validator');
+const jwt = require('jsonwebtoken');
+
+const schema = new passwordValidator();
+
+const PasswordValidator = (password) => {
```
Comment (DaveAvigad) [inline middleware/validatorService.js:7]: keep the convention camelCased variable names

File: middleware/validatorService.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+function generateAccessToken(email) {
+  return jwt.sign(email, process.env.ACCESS_TOKEN_SECRET, { expiresIn: '15m' });
+}
+
+function generateRefreshToken(email) {
+  return jwt.sign(email, process.env.REFRESH_TOKEN_SECRET, { expiresIn: '1d' });
+}
+
+schema
```
Comment (DaveAvigad) [inline middleware/validatorService.js:56]: 🎉

File: package.json [hunk 11]
```diff
@@ -25,16 +26,30 @@
...
+    "eslint": "^8.30.0",
     "express": "^4.18.2",
     "express-session": "^1.17.3",
+    "express-validator": "^6.14.2",
+    "generate-password": "^1.7.0",
     "jsonwebtoken": "^8.5.1",
     "mongoose": "^6.8.0",
     "nodemailer": "^6.8.0",
     "nodemailer-smtp-transport": "^2.7.4",
+    "nodemon": "^2.0.20",
```
Comment (DaveAvigad) [inline package.json:42]: `"nodemon": "^2.0.20",` is a devDep

File: repositories/oneTimePass.repositories.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
+const LocalStorage = require('../data/LocalStorage');
+const MongoStorage = require('../data/MongoStorage');
+
+module.exports = class UsersRepository {
```
Comment (DaveAvigad) [inline repositories/oneTimePass.repositories.js:4]: file name doesn't match the class name

File: routes/usersRoute.js [hunk 13]
```diff
@@ -5,4 +5,10 @@ const usersController = require('../controllers/usersController');
 
 usersRouter.post('/', usersController.getUsers);
```
Comment (ofirpeleg) [inline routes/usersRoute.js:6]: Unnecessary

**Review by DaveAvigad — APPROVED** [review]
looks good 
left some comments

File: controllers/usersController.js [hunk 14]
```diff
@@ -1,3 +1,38 @@
+const { User } = require('../services/authService');
+const { oneTimePass } = require('../services/registerService');
+
 const getUsers = async (req, res) => res.sendStatus(200);
```
Comment (ofirpeleg) [inline controllers/usersController.js:4]: why is this here for?

File: controllers/usersController.js [hunk 15]
```diff
@@ -1,3 +1,38 @@
...
 
-module.exports = { getUsers };
```
Comment (ofirpeleg) [inline controllers/usersController.js]: and this?

File: data/MongoStorage.js [hunk 16]
```diff
@@ -21,7 +21,7 @@ module.exports = class MongoStorage extends EventEmitter {
   }
 
   find() {
-    return this.Model.findOne({});
+    return this.Model.find({});
```
Comment (ofirpeleg) [inline data/MongoStorage.js:24]: 👍

File: routes/usersRoute.js [hunk 17]
```diff
@@ -5,4 +5,10 @@ const usersController = require('../controllers/usersController');
...
 
+usersRouter.get('/', usersController.handleGetUsers);
+usersRouter.get('/:email', usersController.handleGetUser);
+
+usersRouter.put('/:email', usersController.handleUpdateUser);
+usersRouter.delete('/:email', usersController.handleDeleteUser);
+
```
Comment (ofirpeleg) [inline routes/usersRoute.js:13]: 🙌🏻

Comment (ofirpeleg) [reply routes/usersRoute.js:13]: need to add if userExist checking before update/delete/retrive 👾
